src/v0/destinations/sendinblue/transform.js

```diff
@@ -0,0 +1,390 @@
+/* eslint-disable camelcase */
+const { EventType } = require("../../../constants");
+const {
+  defaultRequestConfig,
+  simpleProcessRouterDest,
+  constructPayload,
+  removeUndefinedAndNullValues,
+  defaultPostRequestConfig,
+  TransformationError,
+  getFieldValueFromMessage,
+  getDestinationExternalID,
+  getDestinationExternalIDs,
+  defaultPutRequestConfig
+} = require("../../util");
+const { isNotEmpty, getIntegrationsObj } = require("../../util");
+const { CONFIG_CATEGORIES, MAPPING_CONFIG } = require("./config");
+const { TRANSFORMER_METRIC } = require("../../util/constant");
+const { DESTINATION } = require("./config");
+const {
+  prepareEmailFromPhone,
+  checkIfEmailOrPhoneExists,
+  validateEmailAndPhone,
+  checkIfContactExists,
+  prepareHeader,
+  removeEmptyKey,
+  prepareUserTraits,
+  prepareTrackEventData
+} = require("./util");
+
+const responseBuilder = (
+  payload,
+  endpoint,
+  destination,
+  trackerApi = false,
+  method = defaultPostRequestConfig.requestMethod
+) => {
+  if (payload) {
+    const response = defaultRequestConfig();
+    const { apiKey, clientKey } = destination.Config;
+    response.endpoint = endpoint;
+    response.headers = prepareHeader(apiKey, clientKey, trackerApi);
+    response.method = method;
+    response.body.JSON = removeUndefinedAndNullValues(removeEmptyKey(payload));
+    return response;
+  }
+  // fail-safety for developer error
+  throw new TransformationError(
+    "Something went wrong while constructing the payload",
+    400,
+    {
+      scope: TRANSFORMER_METRIC.MEASUREMENT_TYPE.TRANSFORMATION.SCOPE,
+      meta: TRANSFORMER_METRIC.MEASUREMENT_TYPE.TRANSFORMATION.META.BAD_EVENT
+    },
+    DESTINATION
+  );
+};
+
+// ref:- https://developers.sendinblue.com/reference/createcontact
+const createOrUpdateContactResponseBuilder = (message, destination) => {
+  const { endpoint } = CONFIG_CATEGORIES.CREATE_OR_UPDATE_CONTACT;
+  const payload = constructPayload(
+    message,
+    MAPPING_CONFIG[CONFIG_CATEGORIES.CREATE_OR_UPDATE_CONTACT.name]
+  );
+
+  checkIfEmailOrPhoneExists(payload.email, payload.attributes?.SMS);
+  validateEmailAndPhone(payload.email, payload.attributes?.SMS);
```

Comment:
Can't we merge these two functions?

Reply:
In some places `email` is mandatory where we don't need to check if `phone` is exists in input or not. For that we can skip the use of 
`checkIfEmailOrPhoneExists` and can simply validate email/phone with `validateEmailAndPhone`. Thats why i have created these two separate functions.

Reply:
We can create one function and provide some flags to differentiate between these two functionalities.

Reply:
Yes, 2 separate function with different functionality